src/test/java/com/example/OnlineShopApiApplication/EmployeeServiceTests.java

```diff
@@ -0,0 +1,290 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class EmployeeServiceTests {
+    @InjectMocks
+    EmployeeService employeeService;
+
+    @Mock
+    EmployeeRepository employeeRepository;
+    @Mock
+    RoleRepository roleRepository;
+    @Mock
+    JobTypeRepository jobTypeRepository;
+    @Mock
+    BindingResult bindingResult;
+    @Test
+    void testGetAllEmployeesSuccess() {
+        Role role = Role.builder().id(2L).name("ROLE_EMPLOYEE").build();
+        JobType jobType = JobType.builder().id(2L).name("CASHIER").build();
+        Employee employee = new Employee();
+        employee.setId(1L);
+        employee.setFirstName("John");
+        employee.setRole(role);
+        employee.setJobType(jobType);
+        List<Employee> employees = List.of(employee);
+        when(employeeRepository.findAll()).thenReturn(employees);
+
+        List<EmployeeResponseDto> expectedDtoList = EmployeeMapper.toDtoList(employees);
+
+        ResponseEntity<List<EmployeeResponseDto>> response = employeeService.getAllEmployees();
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(expectedDtoList, response.getBody());
+        verify(employeeRepository, times(1)).findAll();
+    }
+
+    @Test
+    void testGetAllEmployeesException() {
+        when(employeeRepository.findAll()).thenThrow(new RuntimeException("Database error"));
+
+        ResponseEntity<List<EmployeeResponseDto>> response = employeeService.getAllEmployees();
+
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, response.getStatusCode());
+        assertNull(response.getBody());
+        verify(employeeRepository,times(1)).findAll();
+    }
+
+    @Test
+    void testGetAllEmployeesExceptAdminsSuccess() {
+        Role employeeRole = Role.builder().id(2L).name("ROLE_EMPLOYEE").build();
+        Role adminRole = Role.builder().id(1L).name("ROLE_ADMIN").build();
+        JobType jobType = JobType.builder().id(2L).name("CASHIER").build();
+        JobType admin = JobType.builder().id(0L).name("ADMIN").build();
+
+        Employee adminEmployee = new Employee();
+        adminEmployee.setId(1L);
+        adminEmployee.setFirstName("Admin");
+        adminEmployee.setRole(adminRole);
+        adminEmployee.setJobType(admin);
+
+        Employee employee = new Employee();
+        employee.setId(2L);
+        employee.setFirstName("John");
+        employee.setRole(employeeRole);
+        employee.setJobType(jobType);
+
+        List<Employee> employees = List.of(employee, adminEmployee);
```

Comment:
The test testGetAllEmployeesExceptAdminsSuccess will currently return the admin as well, simply because we've mocked the repository to return a list that includes them.